tests/test_contracts.py

```diff
@@ -202,3 +225,12 @@ def test_errback(self):
 
         self.assertFalse(self.results.failures)
         self.assertTrue(self.results.errors)
+
+    @defer.inlineCallbacks
+    def test_same_url(self):
+        TestSameUrlSpider.start_requests = lambda s: self.conman.from_spider(s, self.results)
```

Comment:
if TestSameUrlSpider can be used only in this test (as its start_requests is monkey-patched), what do you think about just defining the spider inside the method?

Reply:
Makes sense to me!